core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt

```diff
@@ -28,9 +30,12 @@ inline fun <reified T : ViewModel> Fragment.viewModel(
   viewModelFactory: ViewModelProvider.Factory
 ) = ViewModelProviders.of(this, viewModelFactory).get(T::class.java)
 
-fun Fragment.toast(
-  stringId: Int,
-  length: Int = Toast.LENGTH_LONG
-) {
+fun Fragment.toast(stringId: Int, length: Int = Toast.LENGTH_LONG) {
   requireActivity().toast(stringId, length)
 }
+
+fun Fragment.closeKeyboard() {
+  val inputMethodManager =
+    requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+  inputMethodManager.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
+}
```

Comment:
Can't we make some extension on context for keyboard-management, this is just Fragment specific

Reply:
What are the pros of having it context specific?

Reply:
`requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager`

this line would get shorter. Really it is about asking yourself what class should have this responsibility. Context might not be a great choice as this method is going to change in light of my other comment and a view will become necessary, notably not all `contexts` actually have views. It may make most sense to be on View or Fragment at the end of the day. View makes slightly more sense but putting it on fragment is easier and more inline with the general usage. Fragments also technically can have no view but we don't use them like that and nobody should use them like that anymore now that ViewModels exist